src/resdk/resources/annotations.py

```diff
@@ -0,0 +1,124 @@
+"""Annotatitons resources."""
+
+import logging
+from typing import TYPE_CHECKING, Optional, Union
+
+from .base import BaseResource
+
+if TYPE_CHECKING:
+    from resdk.resolwe import Resolwe
+
+
+class AnnotationGroup(BaseResource):
+    """Resolwe AnnotationGroup resource."""
+
+    # There is currently no endpoint for AnnotationGroup object, but it might be
+    # created in the future. The objects are created when AnnotationField is
+    # initialized.
+    endpoint = "annotation_group"
+
+    READ_ONLY_FIELDS = BaseResource.READ_ONLY_FIELDS + ("name", "sort_order", "label")
+
+    def __init__(self, resolwe: "Resolwe", **model_data):
+        """Initialize the instance.
+
+        :param resolwe: Resolwe instance
+        :param model_data: Resource model data
+        """
+        self.logger = logging.getLogger(__name__)
+        super().__init__(resolwe, **model_data)
+
+
+class AnnotationField(BaseResource):
+    """Resolwe AnnotationField resource."""
+
+    endpoint = "annotation_field"
+
+    READ_ONLY_FIELDS = BaseResource.READ_ONLY_FIELDS + (
+        "description",
+        "group",
+        "label",
+        "name",
+        "sort_order",
+        "type",
+        "validator_regex",
+        "vocabulary",
+        "required",
+    )
+
+    def __init__(self, resolwe: "Resolwe", **model_data):
+        """Initialize the instance.
+
+        :param resolwe: Resolwe instance
+        :param model_data: Resource model data
+        """
+        self.logger = logging.getLogger(__name__)
+        #: annotation group
+        self._group = None
+        super().__init__(resolwe, **model_data)
+
+    @property
+    def group(self) -> AnnotationGroup:
+        """Get annotation group."""
+        assert (
+            self._group is not None
+        ), "AnnotationGroup must be set before it can be used."
+        return self._group
+
+    @group.setter
+    def group(self, payload: dict):
+        """Set annotation group."""
+        self._resource_setter(payload, AnnotationGroup, "_group")
+
+
+class AnnotationValue(BaseResource):
+    """Resolwe AnnotationValue resource."""
+
+    endpoint = "annotation_value"
+
+    READ_ONLY_FIELDS = BaseResource.READ_ONLY_FIELDS + ("label",)
+
+    UPDATE_PROTECTED_FIELDS = BaseResource.UPDATE_PROTECTED_FIELDS + ("entity", "field")
```

Comment:
The filed `entity` is called sample in resdk. Could you enable that users would be able to relate to the sample as `AnnotationValue.sample`? However, this will only return Sample ID. So maybe it is better to expose this as `AnnotationValue.sample_id`... We can discuss this live.